lib/auth/init.go

```diff
@@ -586,6 +586,17 @@ func checkResourceConsistency(keyStore keystore.KeyStore, clusterName string, re
 			if err := checkRoleFeatureSupport(r); err != nil {
 				return trace.Wrap(err)
 			}
+		case types.GithubConnector:
+			// If we are running Teleport OSS, ensure that the Github organization
+			// the user is trying to authenticate with is not using external SSO.
+			// SSO is a Teleport Enterprise feature and shouldn't be allowed in OSS.
+			// This is checked when Github auth connectors get created or updated, but
+			// check again here in case the organization enabled external SSO after
+			// the auth connector was created.
+			if err := checkGithubFeatureSupport(r); err != nil {
+				return trace.Wrap(err)
```

Comment:
If one day github stops properly handling requests for the `sso` page and we get some network error rather than an HTTP error code, we really don't want to brick every teleport cluster using github SSO. So perhaps if the error is not `AccessDenied`, we should just log a warning and continue. I think that would be the safer option, given the pretty big risk.

Reply:
Why are we checking this in init? I don't think this is an appropriate place for it.

Reply:
@r0mant That was my suggestion, I just thought there's no reason to call out to github on every login if we can check it when you upsert the connector and again during init (where we check license support for all other items in the backend). Are you concerned we'll break init, or we won't catch some people who turned on SSO after starting up their cluster?

Reply:
Both :) I just think checking in the actual callback is the most reliable and least error-prone. And like you said, if Github is down, we either break startup or skip this check in which is it's basically a no-op (and can trivially be bypassed by e.g. blocking Github IPs on startup and then unblocking them).

Moreover, right now it's placed in `checkResourceConsistency` which is only called on the first start which doesn't account for restarts and upgrades.

Reply:
Makes sense. I didn't realize this doesn't run on restarts tbh

Reply:
this is all OSS though, if you're willing to block IPs during startup there's nothing stopping you from just deleting this function :)

Reply:
Ok cool, I'll go ahead and re-add checks when logging in